python/lsst/ts/watcher/rules/cp_verify_alarm.py

```diff
@@ -0,0 +1,305 @@
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["CpVerifyAlarm"]
+
+import yaml
+import json
+import lsst.daf.butler as dafButler
+import salobj
+import asyncio
+import concurrent.futures
+
+from lsst.ts.idl.enums.Watcher import AlarmSeverity
+from lsst.ts import watcher
+
+
+class CpVerifyAlarm(watcher.BaseRule):
+    """Monitor state of cp_verify executions from the OCPS
+    for a particular type of calibration (BIAS, DARK, FLAT, etc).
+
+    Set alarm severity NONE if the cp_verify tests passed,
+    and issue a WARNNING  alarm if not.
+
+    Parameters
+    ----------
+    config : `types.SimpleNamespace`
+        Rule configuration, as validated by the schema.
+
+    Notes
+    -----
+    The alarm name is f"CpVerifyAlarm.{ocps_index}:{calibration_type}",
+    where ``calibration_type`` and ``ocps_index`` are derived from
+    ``config.calibration_type`` and ``config.ocps_index``, respectively.
+    """
+
+    def __init__(self, config):
+        remote_name = "OCPS"
+        remote_index = config.ocps_index
+        remote_info = watcher.RemoteInfo(
+            name=remote_name,
+            index=remote_index,
+            callback_names=["job_result"],
+            poll_names=[],
+        )
+        super().__init__(
+            config=config,
+            name=f"CpVerifyAlarm.{remote_info.index}:{remote_info.name}",
+            remote_info_list=[remote_info],
+        )
+
+        self.calibration_type = config.calibration_type
+
+    @classmethod
+    def get_schema(cls):
+        schema_yaml = """
+            $schema: http://json-schema.org/draft-07/schema#
+            description: Configuration for CpVerifyBiasAlarm
+            type: object
+            properties:
+                calibration_type:
+                    description: >-
+                        Alarm calibration type: BIAS, DARK, or FLAT.
+                    type: string
+                    enum: ['BIAS', 'DARK', 'FLAT']
+                ocps_index:
+                    type: integer
+                    enum: [1, 2]
+                    description: >-
+                        OCPS index (e.g., OCPS:1: LATISS; OCPS:2: LSSTComCam).
+                verification_threshold:
+                    type: integer
+                    description: Maximum number of failures per detector per test type.
+                default: 8
+            required: [calibration_type, ocps_index]
+            additionalProperties: false
+        """
+        return yaml.safe_load(schema_yaml)
+
+    def __call__(self, topic_callback):
+        msg = topic_callback.get()
+        # OCPS response after calling the pipetask
+        response = json.loads(msg.result)
+        # Get the dictionary with cp_verify stats, from the butler.
+        # It might be slow, so use `run_in_executor`
+        loop = asyncio.get_running_loop()
+        verify_stats = await loop.run_in_executor(concurrent.futures.ThreadPoolExecutor(max_workers=3),
```

Comment:
> Pass the argument(s) for get_cp_verify as additional arguments to run_in_executor.
Like this?: `verify_stats = await loop.run_in_executor(concurrent.futures.ThreadPoolExecutor(max_workers=3),
                                                  self.get_cp_verify_stats, ocps_result`

Reply:
> If you are confident that these tasks will never build up, then I'd pass None. Otherwise I'd construct the executor once.

I have passed `None`. That function, `get_cp_verify_stats`, is parsing the OCPS result file from running `cp_verify` in order to identify 1) the calibration type and 2) call the butler with the appropriate verification collection (whose name is formed from the OCPS job ID) to get the dictionary that has the verification statistics we want to check. Unless I'm missing something, I wouldn't think this butler call will, a priori, be slow. 

`        verify_stats = await loop.run_in_executor(None,
                                                  self.get_cp_verify_stats, ocps_result)`